src/shared/tpm2-util.c

```diff
@@ -2333,3 +2350,62 @@ int pcr_mask_to_string(uint32_t mask, char **ret) {
         *ret = TAKE_PTR(buf);
         return 0;
 }
+
+#define PBKDF2_HMAC_SHA256_ITERATIONS 10000
+
+/*
+ * Implements PBKDF2 HMAC SHA256 for a derived keylen of 32
+ * bytes and for PBKDF2_HMAC_SHA256_ITERATIONS count.
+ * I found the wikipedia entry relevant and it contains links to
+ * relevant RFCs:
+ *   - https://en.wikipedia.org/wiki/PBKDF2
+ *   - https://www.rfc-editor.org/rfc/rfc2898#section-5.2
+ */
+int tpm2_util_pbkdf2_hmac_sha256(const void *pass,
+                    size_t passlen,
+                    const void *salt,
+                    size_t saltlen,
+                    uint8_t res[static SHA256_DIGEST_SIZE]) {
```

Comment:
in the systemd codebase we usually name return parameters `ret`or `ret_somesuffix`. Hence i think it would make sense to rename "res" to "ret" here